Devnot.Mentor.Api/Controllers/MentorController.cs

```diff
@@ -30,6 +30,24 @@ public async Task<IActionResult> GetAsync([FromRoute] string userName)
             return result.Success ? Success(result) : BadRequest(result);
         }
 
+        [HttpGet]
+        [Route("/mentors/{userName}/mentees")]
+        public async Task<IActionResult> GetMentees([FromRoute] string userName)
+        {
+            var result = await mentorService.GetMentees(userName);
+
+            return result.Success ? Success(result) : BadRequest(result);
+        }
+
+        [HttpGet]
+        [Route("/mentors/{userName}/applications")]
+        public async Task<IActionResult> GetApplications([FromRoute] string userName)
+        {
+            var result = await mentorService.GetApplicationsNotIncludeApproveds(userName);
+
+            return result.Success ? Success(result) : BadRequest(result);
+        }
```

Comment:
Burada direkt parametre olarak gelen kullanıcı adına göre mentore ait onaylanan öğrenci ve başvuru listesi getiriliyor. Mentor'un destek verdiği öğrenci ve aldığı başvuruları direkt göstermek uygun değil gibi. Bir X kişisi Y kişisine ait destek verdiği öğrenci ve aldığı başvuruları görebiliyor bu durumda, her ne kadar X'e ait token olsa bile X == Y gibi kontrolleri gerçekleştirmiyoruz. 

Route kısmı **/users/{userId}/mentor/applications** gibi bir desene sahip olsa ve **userId** bilgisi de token içerisinden alınsa daha sağlıklı olur gibi geliyor bana. Hem güvenlik sağlanmış olur hem de başkasına ait verilere direkt erişilmez diye düşünüyorum.

Reply:
ben 
- öğrencinin başvuru yaptığı mentorları ve anlık olarak mentorluk desteği aldığı mentorları
- mentorun da aynı şekilde başvuru aldığı öğrencileri ve anlık olarak mentorluk desteği verdiği öğrencileri

öğrencinin veya mentorun profiline bakıldığında halka açık olarak sunabileceğimizi düşündüm.

ilk önce, bunların nasıl şekilleneceğine karar verelim. sonraysa istenilen düzeltmeleri tekrar yapabilirim.

Reply:
Token içerisinde UserId değeri var diye Route kısmını UserId olarak düzelttim. `Login` metotu sonrasında token içerisinde kullanıcı adı bilgisini de koyabilirsin istersen. Route değişmemiş de olur.

Reply:
Evet, o kısmı belirtmemiz gerekiyordu. Şu anlık gösterme gibi bir düşüncemiz yok açıkçası. İleride opsiyonel bir hale getirilebilir.

Reply:
Tamamdır öyleyse, bu konuda yapılması gerekenler: yukarıda bahsettiğimiz başvuru durumları, aktif mentorluk veya bitmiş süreçleri sadece authorize olmuş ve bağlantısı olan isteklere sunacağız. Route pattern'i ise `users` altında bulunacak.

Doğru anladığımı onaylarsanız, bu konularla alakalı geliştirmeleri tekrar yaparım.

Reply:
Evet, aynı işlemler. Talebi gönderen kişiye ait bilgiler gelecek. Route kısmında ise illa user olması şart değil. Böyle bir route de olabilir: /mentors/**me**/applications gibi. userName yerine me kelimesi gelebilir zaten istek gönderen kişiye ait olacağı için. Token içerisinde de UserId bilgisi yer alıyor.